common/presenters/allocations-to-table-component.js

```diff
@@ -0,0 +1,129 @@
+const i18n = require('../../config/i18n')
+const filters = require('../../config/nunjucks/filters')
+const componentService = require('../services/component')
+
+const tablePresenters = require('./table')
+
+function tagByRemaining(totalSlots, unfilledSlots) {
+  const isComplete = unfilledSlots === 0
+  const inProgress = unfilledSlots !== totalSlots
+  const text = i18n.t('collections::labels.progress_status', {
+    context: inProgress ? (isComplete ? 'complete' : 'remaining') : '',
```

Comment:
`const context = inProgress ? (isComplete ? 'complete' : 'remaining') : ''`

(Still find that a bit of an overloaded double ternary)

and then this just provide an object to map the classes

```
const contextClasses = {
  complete: 'govuk-tag--green',
  remaining: 'govuk-tag-yellow',
  default: 'govuk-tag--red'
}
...
const classes = contextClasses[context] || contextClasses.default
```
(or similar)

Reply:
That does read much nicer. And then can actually move that out of those methods to avoid duplication.